CI with the latest stable(GA) version of MariaDB 10.2

yahonda opened the pull request:
### Summary
This pull request updates MariaDB version to 10.2 since #29708 and #29706 supports MariaDB 10.2 changes.

There was a similar pull request #29740 which was closed since MariaDB 10.2.6 GA is not supported on Ubuntu 12.04 LTS.

Now Travis CI is migrating Ubuntu version to Trusty and  MariaDB 10.2 is supported on Ubuntu Trusty I think it is good timing to open another pull request.

Refer

* https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming

* https://downloads.mariadb.org/mariadb/repositories/#mirror=digitalocean-nyc&distro=Ubuntu&distro_release=trusty--ubuntu_trusty&version=10.2

yahonda committed:
CI with the latest stable(GA) version of MariaDB 10.2

- Travis CI is migrating Ubuntu version to Trusty
- MariaDB 10.2 is supported on Ubuntu Trusty

yahonda commented:
I'm wondering if it actually installs MariaDB 10.2. 

https://travis-ci.org/rails/rails/jobs/255286603

```ruby
$ mysql --version
mysql  Ver 14.14 Distrib 5.6.33, for debian-linux-gnu (x86_64) using  EditLine wrapper
```

eugeneius commented:
Installing MariaDB 10.2 on Travis is currently broken: https://github.com/travis-ci/travis-build/pull/1091

yahonda commented:
https://github.com/travis-ci/travis-build/pull/1153 addresses install MariaDB 10.2 on Travis CI.

I have pushed my commit and confirmed MariaDB 10.2.8 is running here https://travis-ci.org/rails/rails/jobs/270397972#L529-L530

```sql
$ mysql --version
mysql  Ver 15.1 Distrib 10.2.8-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
```

However, it gets an error. Opened #30485 with steps to reproduce.

```ruby
Error:
ActiveRecord::Migration::ColumnsTest#test_remove_column_with_multi_column_index:
ActiveRecord::StatementInvalid: Mysql2::Error: Key column 'hat_size' doesn't exist in table: ALTER TABLE `test_models` DROP `hat_size`
```

yahonda committed:
Skip `test_remove_column_with_multi_column_index`
when tested with MariaDB 10.2.8 or higher

Refer #30485

https://mariadb.com/kb/en/the-mariadb-library/alter-table/#drop-column-if-exists-col_name-cascaderestrict
> MariaDB starting with 10.2.8
> Dropping a column that is part of a multi-column UNIQUE constraint is not permitted.

yahonda commented:
Now build is green.

* Travis CI supports MariaDB 10.2 on Ubuntu Trusty now.
* MariaDB 10.2.8 or later does not allow to drop a column which is part of multi column unique constraint, this pull request skip `test_remove_column_with_multi_column_index` if MariaDB 10.2.8 or higher.

kamipo commented on activerecord/test/cases/migration/columns_test.rb:
```diff
@@ -142,6 +142,7 @@ def test_remove_column_with_index
       end
 
       def test_remove_column_with_multi_column_index
+        skip if current_adapter?(:Mysql2Adapter) && @connection.mariadb? && @connection.version >= "10.2.8"
```
Can you add a comment on the line why we need to skip for the MariaDB version?

yahonda replied:
Comment added and squashed.

kamipo commented on activerecord/test/cases/migration/columns_test.rb:
```diff
@@ -142,6 +142,10 @@ def test_remove_column_with_index
       def test_remove_column_with_multi_column_index
+        # MariaDB starting with 10.2.8
+        # Dropping a column that is part of a multi-column UNIQUE constraint is not permitted.
+        skip if current_adapter?(:Mysql2Adapter) && @connection.mariadb? && @connection.version >= "10.2.8"
```
`@connection` is only used in this line in the file.
I prefer to use `connection` rather than `@connection`.

yahonda replied:
Understood. I have another question. 
I'm wondering `if current_adapter?(:Mysql2Adapter) && connection.mariadb?` is redundant
since `mariadb?` is only available in mysql2 adapter. I have no strong preference but wanted to hear from your opinion.

kamipo replied:
I agree with you, but we have no `connection.mariadb?` in test cases for now, so I'm acceptable that.

```
% git grep -n -i 'mariadb' test
test/cases/adapters/mysql2/datetime_precision_quoting_test.rb:22:  test "microsecond precision for MariaDB gte 5.3.0" do
test/cases/adapters/mysql2/datetime_precision_quoting_test.rb:23:    stub_version "5.5.5-10.1.8-MariaDB-log" do
test/cases/adapters/mysql2/datetime_precision_quoting_test.rb:28:  test "no microsecond precision for MariaDB lt 5.3.0" do
test/cases/adapters/mysql2/datetime_precision_quoting_test.rb:29:    stub_version "5.2.9-MariaDB" do
```

yahonda replied:
Thanks for the reply. Modified to use `connection` here and keeping `if current_adapter?(:Mysql2Adapter) && connection.mariadb?` condition.
